### ISSUE TITLE
segfaults releated to loading a core

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#3702]
Cores allow to come up with an own WML tree. There are some things which the games assumes to exist, but doesn't have to, and segfaults:

- [ ] not having a single era and entering MP (not having campaigns and clicking campaigns yields anm error message instead)
- [ ] not having a single MP scenario and entering MP. Ideally, it should look if you have one of the Random scenarios, an MP campaign or a custom map. Note that specifying the section for custom maps is done by the core, not the engine. (https://github.com/wesnoth/wesnoth/blob/1.14/data/multiplayer/_main.cfg#L6-L13)
- [ ] opening the preferences but not sourcing {advanced_preferences.cfg} (or in other words,having no [advanced_preference] tags)
- [ ] as well related to cores: https://bpaste.net/show/4702a765bcdf

probably more into which I didn't run

**Comment by CelticMinstrel** [comment]
* I'd argue the advanced preferences should probably be moved out of the main WML and into a separate WML tree (at least partially), as they seem to be largely things that shouldn't depend on the core.
* Help on the other hand... I'd say, just hide the Help button in the titlescreen if the help config is missing.
* For MP it should probably behave similarly to campaigns, with an error message disallowing you from entering the mode at all.

**Comment by sevu** [comment]
One can as well insert his own tutorial … but maybe the button for it should just go if one doesn't? (It has an error message currently)

**Comment by CelticMinstrel** [comment]
Yeah, I'd say the tutorial button should be hidden if there is no tutorial.

Bonus points if it puts up a choice dialog (similar to the test-scenarios hotkey) if there are more than one tutorial scenario with `allow_new_game=yes`,
